### PR TITLE
add tests and validations

### PR DESCRIPTION
**Pull request by MateeSaafi**
# Project requirements
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+    it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
+  end
+
+  describe 'associations' do
+    it { should belong_to(:author).class_name('User') }
+    it { should have_many(:comments).dependent(:destroy) }
+    it { should have_many(:likes).dependent(:destroy) }
+  end
+end
```
Comment (DeliceLydia) [inline spec/models/post_spec.rb]: - [ ] Please fix tests that are failing
Screen Shot 2023-06-24 at 21 33 13

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @matee-safi, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Added validations
- Followed gitflow


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (Whoistolu) [reply spec/models/post_spec.rb]: - [ ] Well resolved 👍🏽 
Screenshot 2023-06-24 at 21 05 33

**Review by Whoistolu — APPROVED** [review]
### Hi @matee-safi 

Great job implementing the recommended changes 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**